### PR TITLE
Solved issue #4187- Tower Defence Game

### LINKED ISSUE #4187: [New game]: Tower defence
### 🎮 Game Request

Game logic and basic description
In this game, players place defensive towers along a path to stop waves of enemies from reaching the end. Each tower has different abilities and ranges, and players can upgrade towers to improve their effectiveness.

### Point down the features

Game points
Multiple Tower Types:

Basic Tower: Standard tower with medium range and attack speed.
Sniper Tower: Long-range tower with high damage but slow attack speed.
Rapid Fire Tower: Short-range tower with low damage but very fast attack speed.
Splash Damage Tower: Deals area-of-effect damage to multiple enemies at once.
Enemy Variety:

Basic Enemy: Standard enemy with average speed and health.
Fast Enemy: High-speed enemy with lower health.
Tank Enemy: Slow-moving enemy with high health.
Flying Enemy: Can only be attacked by certain towers.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by Aditi22Bansal**
## PR Description 📜

Please include summary related to the issue you have fixed and describe your PR in brief over here by specifying the issue number on which you were working below
In this game, players place defensive towers along a path to stop waves of enemies from reaching the end. Each tower has different abilities and ranges, and players can upgrade towers to improve their effectiveness.

Fixes # <your_issue_number>
Fixes #4187 

<hr>
 
## Mark the task you have completed ✅

- [ ] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [x] I have added screenshot for website preview in assets/images 
- [x] I have added entries for my game in main README.md
- [x] I have added README.md in my folder 
- [x] I have added working video of the game in README.md (optional)
- [x] I have specified the respective issue number for which I have requested the new game.

<hr>

## Add your screenshots(Optional) 📸
Screenshot 2024-06-04 004952
Screenshot 2024-06-04 005003
Screenshot 2024-06-04 005020

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @Aditi22Bansal ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review and merge

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @Aditi22Bansal,
You need to make the following changes:
  - Please add an image to the assets/images directory.

Hoping that you will make those changes soon 🚀

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly check I have added

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Done

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review and merge

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review and merge

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review and merge
Thankyou

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review and merge

**Comment by Aditi22Bansal** [comment]
@ishita-43 
Kindly review this PR

**Comment by Aditi22Bansal** [comment]
@ishita-43 
Kindly review this PR so that I can work on the other issue you have assigned me

**Review by ishita-43 — CHANGES_REQUESTED** [review]
Please make the following changes:- ⚠ 
Follow contribution guidelines while naming the images in assets folder
Do not place images outside of your game folder instead place it in your game folder.

i hope you will make changes soon!!

**Comment by Aditi22Bansal** [comment]
ok

**Comment by Aditi22Bansal** [comment]
@ishita-43 
I have done the required changes
Kindly review 
If any changes are required kindly tell

File: Screenshot 2024-06-04 005020.png
Comment (ishita-43) [inline Screenshot 2024-06-04 005020.png:1]: remove this one..

File: Tower game.png
Comment (ishita-43) [inline Tower game.png:1]: remove this one..

File: assets/images/Tower_Game.png
Comment (ishita-43) [inline assets/images/Tower_Game.png:1]: name as "Tower_Defence_Game.png"

**Review by ishita-43 — CHANGES_REQUESTED** [review]
Also change the path of images if used anywhere in the program

**Pull request closed without merging**